HerPortal.ManagementShell/AdminAction.cs

```diff
@@ -32,161 +35,87 @@ public AdminAction(IDatabaseOperation dbOperation, IOutputProvider outputProvide
     {
         return userOrNull == null ? UserStatus.New : UserStatus.Active;
     }
 
-    private void TryCreateUser(string userEmailAddress, string[]? custodianCodes)
+    public void CreateUser(string userEmailAddress, IReadOnlyCollection<string>? custodianCodes,
+        IReadOnlyCollection<string>? consortiumCodes)
     {
         var lasToAdd = dbOperation.GetLas(custodianCodes ?? Array.Empty<string>());
-        dbOperation.CreateUserOrLogError(userEmailAddress, lasToAdd);
+        var consortiaToAdd = dbOperation.GetConsortia(consortiumCodes ?? Array.Empty<string>());
+
+        dbOperation.CreateUserOrLogError(userEmailAddress, lasToAdd, consortiaToAdd);
     }
 
-    public void TryRemoveUser(User? user)
+    public void RemoveUser(User user)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        var deletionConfirmation = outputProvider.Confirm(
-            $"Attention! This will delete user {user.EmailAddress} and all associated rows from the database. Are you sure you want to commit this transaction? (y/n)");
-        if (!deletionConfirmation)
-        {
-            return;
-        }
-
         dbOperation.RemoveUserOrLogError(user);
     }
-    private void AddLas(User? user, string[]? custodianCodes)
+
+    public void AddLas(User user, IEnumerable<string> custodianCodes)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        if (custodianCodes == null || custodianCodes.Length < 1)
-        {
-            outputProvider.Output("Please specify custodian codes to add to user");
-            return;
-        }
-
-        var lasToAdd = dbOperation.GetLas(custodianCodes);
+        var filteredCustodianCodes = custodianCodes
+            .Where(custodianCode => !CustodianCodeIsInOwnedConsortium(user, custodianCode))
+            .ToList();
+
+        var lasToAdd = dbOperation.GetLas(filteredCustodianCodes);
+
         dbOperation.AddLasToUser(user, lasToAdd);
     }
 
-    public void RemoveLas(User? user, string[]? custodianCodes)
+    public void RemoveLas(User user, IReadOnlyCollection<string> custodianCodes)
     {
-        if (user == null)
-        {
-            outputProvider.Output("User not found");
-            return;
-        }
-
-        if (custodianCodes == null || custodianCodes.Length < 1)
-        {
-            outputProvider.Output("Please specify custodian codes to remove from user");
-            return;
-        }
-
-        var userConfirmation = ConfirmCustodianCodes(user.EmailAddress, custodianCodes);
-        if (!userConfirmation)
-        {
-            return;
-        }
-
         var lasToRemove = user.LocalAuthorities.Where(la => custodianCodes.Contains(la.CustodianCode)).ToList();
         var missingCodes = custodianCodes.Where(code => !lasToRemove.Any(la => la.CustodianCode.Equals(code))).ToList();
```

Comment:
Can we use `Contains` here while we're at it?

Reply:
I don't think directly; since the logic is to check whether the list contains an LA with the code

I could do a select -> contains if that seems more natural?

Reply:
think it would be yes